spock-specs/src/test/groovy/org/spockframework/smoke/DisambiguationInConditions.groovy

```diff
@@ -0,0 +1,103 @@
+package org.spockframework.smoke
+
+import spock.lang.Specification
+
+import java.util.regex.Pattern
+
+import static DisambiguationInConditionsTestee.Java1
+import static DisambiguationInConditionsTestee.Java2
+import static DisambiguationInConditionsTestee.Java3
+import static DisambiguationInConditionsTestee.Java4
+import static DisambiguationInConditionsTestee.Verifier
+
+class DisambiguationInConditions extends Specification {
```

Comment:
We probably need new tests in the `org.spockframework.smoke.condition.ConditionRendering` Spec or a related one

Reply:
You think so?
I didn't touch any rendering.

Reply:
This comment was based on my wrong assumption above, that the `result` was not used in the cast case.